skglm/solvers/cd_solver.py

```diff
@@ -111,17 +105,13 @@ def cd_solver_path(X, y, datafit, penalty, alphas=None,
         # alphas = np.sort(alphas)[::-1]
 
     n_alphas = len(alphas)
-
-    coefs = np.zeros((n_features, n_alphas), order='F', dtype=X.dtype)
+    coefs = np.zeros((n_features + fit_intercept, n_alphas), order='F', dtype=X.dtype)
     stop_crits = np.zeros(n_alphas)
 
     if return_n_iter:
         n_iters = np.zeros(n_alphas, dtype=int)
 
     for t in range(n_alphas):
-
-        alpha = alphas[t]
-        penalty.alpha = alpha  # TODO this feels it will break sklearn compat
```

Comment:
I don't see where `alpha` is changed in the case of a path computation, is it working just because we always call it with a single value?

Reply:
Yes indeed, it breaks when fitting with multiple alphas